File: core/src/com/nopalsoft/sharkadventure/handlers/GameServicesHandler.java
package com.nopalsoft.sharkadventure.handlers;

public interface GameServicesHandler {

    void submitScore(long score);

    void unlockAchievement(String achievementId);

    void getLeaderboard();

    void getAchievements();

    boolean isSignedIn();

    void signIn();

}

File: core/src/com/nopalsoft/sharkadventure/handlers/GameServicesHandler.kt
package com.nopalsoft.sharkadventure.handlers

interface GameServicesHandler {
    fun submitScore(score: Long)

    fun unlockAchievement(achievementId: String)

    val leaderboard: Unit

    val achievements: Unit

    val isSignedIn: Boolean

    fun signIn()
}
